trunk/src/app/srs_app_source.cpp

```diff
@@ -855,6 +855,13 @@ srs_error_t SrsOriginHub::initialize(SrsLiveSource* s, SrsRequest* r)
     req_ = r;
     source = s;
     
+    int port; string ip;
+    std::vector<std::string> listens = _srs_config->get_listens();
+    if(listens.size() > 0){
+	srs_parse_endpoint(listens[0], ip, port);
+	req->port = port;
```

Comment:
Please check the CR is failed： https://github.com/ossrs/srs/runs/5241353852?check_suite_focus=true

And, why change the port to listen?